src/System.Memory/src/System/Buffers/SequenceReader_search.cs

```diff
@@ -0,0 +1,798 @@
+                            ;
+                        if (escapeCount == index && priorEscape)
+                        {
+                            // Started and ended with escape, increment once more
+                            escapeCount++;
+                        }
+
+                        priorEscape = false;
+                        if (escapeCount % 2 != 0)
+                        {
+                            // We're in the escaped state, so skip this delimiter
+                            Advance(index + 1);
+                            remaining = UnreadSpan;
+                            continue;
+                        }
+                    }
+
+                    // Found the delimiter. Move to it, slice, then move past it.
+                    if (index > 0)
+                    {
+                        Advance(index);
+                    }
+
+                    sequence = Sequence.Slice(copy.Position, Position);
+                    if (advancePastDelimiter)
+                    {
+                        Advance(1);
+                    }
+                    return true;
+                }
+
+                // No delimiter, need to check the end of the span for odd number of escapes then advance
+                {
+                    int escapeCount = 0;
+                    for (int i = remaining.Length; i > 0 && remaining[i - 1].Equals(delimiterEscape); i--, escapeCount++)
+                        ;
+                    if (priorEscape && escapeCount == remaining.Length)
+                    {
+                        escapeCount++;
+                    }
+                    priorEscape = escapeCount % 2 != 0;
+                }
+
+                // Nothing in the current span, move to the end, checking for the skip delimiter
+                Advance(remaining.Length);
+                remaining = CurrentSpan;
+            }
+
+            // Didn't find anything, reset our original state.
+            this = copy;
+            sequence = default;
+            return false;
+        }
+
+        /// <summary>
+        /// Try to read everything up to the given <paramref name="delimiters"/>.
+        /// </summary>
+        /// <param name="span">The read data, if any.</param>
+        /// <param name="delimiters">The delimiters to look for.</param>
+        /// <param name="advancePastDelimiter">True to move past the first found instance of any of the given <paramref name="delimiters"/>.</param>
+        /// <returns>True if any of the the <paramref name="delimiters"/> were found.</returns>
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public bool TryReadToAny(out ReadOnlySpan<T> span, ReadOnlySpan<T> delimiters, bool advancePastDelimiter = true)
+        {
+            ReadOnlySpan<T> remaining = UnreadSpan;
+            int index = delimiters.Length == 2
+                ? remaining.IndexOfAny(delimiters[0], delimiters[1])
+                : remaining.IndexOfAny(delimiters);
```

Comment:
Is this still necessary given https://github.com/dotnet/coreclr/pull/20738?

Reply:
I can definitely yank it out when the change comes through. This API is critical to Kestrel.